### PR TITLE
fix(ocs-operator): removes use of storageClassDeviceSet PVC for mons

### PR DESCRIPTION
**Pull request by ashishranjan738**
This commit removes the logic of storageClassDeviceSet PVC for
mons, because this doesn't allow using hostPath for mons.

Signed-off-by: Ashish Ranjan <ashishranjan738@gmail.com>

### CODE REVIEW

**Comment by jarrpa** [comment]
Please test your code by running the ocs-operator-ci locally before pushing to GitHub. 🙂

File: pkg/controller/storagecluster/reconcile.go [hunk 1]
```diff
@@ -739,23 +738,8 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string) *cephv1.CephClus
...
 	}
 
-	// If a MonPVCTemplate is provided, use that. If not, if StorageDeviceSets
-	// have been provided, use the StorageClass of the DataPVCTemplate from the
-	// first StorageDeviceSet for providing the Mon PVs
 	if sc.Spec.MonPVCTemplate != nil {
 		cephCluster.Spec.Mon.VolumeClaimTemplate = sc.Spec.MonPVCTemplate
-	} else if len(sc.Spec.StorageDeviceSets) > 0 {
-		ds := sc.Spec.StorageDeviceSets[0]
-		cephCluster.Spec.Mon.VolumeClaimTemplate = &corev1.PersistentVolumeClaim{
```
Comment (jarrpa) [inline pkg/controller/storagecluster/reconcile.go]: Is simply not specifying a PVC template sufficient to tell Rook-Ceph to use hostPath, or is other configuration required?

**Comment by ashishranjan738** [comment]
> Please test your code by running the ocs-operator-ci locally before pushing to GitHub. 🙂

yeah I forgot to fix the integration test.

File: pkg/controller/storagecluster/reconcile.go (hunk 1 above)
Comment (ashishranjan738) [reply pkg/controller/storagecluster/reconcile.go]: I think is sufficient

File: pkg/controller/storagecluster/reconcile.go [hunk 2]
```diff
@@ -739,23 +738,8 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string) *cephv1.CephClus
 		},
 	}
 
-	// If a MonPVCTemplate is provided, use that. If not, if StorageDeviceSets
-	// have been provided, use the StorageClass of the DataPVCTemplate from the
-	// first StorageDeviceSet for providing the Mon PVs
 	if sc.Spec.MonPVCTemplate != nil {
 		cephCluster.Spec.Mon.VolumeClaimTemplate = sc.Spec.MonPVCTemplate
-	} else if len(sc.Spec.StorageDeviceSets) > 0 {
```
Comment (travisn) [inline pkg/controller/storagecluster/reconcile.go]: In addition to removing use of the StorageDeviceSet, we should expose the dataDirHostPath setting from the StorageCluster CR. We can't always assume `/var/lib/rook` is the right value for `dataDirHostPath` as seen above on line 714.

Comment (ashishranjan738) [reply pkg/controller/storagecluster/reconcile.go]: yeah will add this

Comment (ashishranjan738) [reply pkg/controller/storagecluster/reconcile.go]: done

**Comment by ashishranjan738** [comment]
> So thus far I'm seeing the following required changes:
> 
> * Expose the actual hostPath value for configuration
> * Add a unit test for configuring the hostPath
> * Add a unit test for configuring the PVC template

update the PR

**Comment by ashishranjan738** [comment]
@jarrpa can you review this ?

**Pull request merged**